test/common/config/registry_test.cc

```diff
@@ -103,6 +103,17 @@ TEST(RegistryTest, DEPRECATED_FEATURE_TEST(WithDeprecatedFactoryPublished)) {
                           ->name());
 }
 
+class TestWithDeprecatedPublishedFactoryNoDeprecatedNames : public PublishedFactory {
+public:
+  std::string name() const override { return ""; }
```

Comment:
Can this be made to return a non-empty string? It's not clear that the error below isn't caused by this being an invalid value. Maybe the regex should be a bit more precise too.

Reply:
If it's non-empty we don't end up taking the branch that asserts deprecated names are given to the constructor. The constructor is trying to make sure it makes a call to `FactoryRegistry<Base>::registerFactory()`. 

I added a comment as to make it clearer what the test that uses it does.